src/unicorn/bitblasting.rs

```diff
@@ -976,6 +976,82 @@ impl<'a> BitBlasting<'a> {
         }
     }
 
+    fn get_power_two(&mut self, k: &[GateRef]) -> Vec<GateRef> {
+        // computes 2^k
+        if get_non_constant_gate(k).is_none() && get_non_constant_gate(k).is_none() {
```

Comment:
nit-pick: This seems to be the same predicate twice. Would calling it once suffice here?